client/my-sites/sidebar/publish-menu.jsx

```diff
@@ -86,6 +87,7 @@ const PublishMenu = React.createClass( {
 				queryable: true,
 				config: 'manage/media',
 				link: '/media',
+				buttonLink: '/media/' + site.slug,
```

Comment:
Ah, and it looks like testing http://calypso.localhost:3000/media/:siteId: with a clean cache comes up with a number of errors:
<img width="897" alt="screen shot 2017-03-01 at 12 12 49 pm" src="https://cloud.githubusercontent.com/assets/1270189/23479155/68894dca-fe78-11e6-87eb-fa3b936ca54a.png">

To clean out the cache on chrome or ff:
```js
localStorage.clear(); indexedDB.deleteDatabase( 'calypso' );
```